wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/EditingResult.java

```diff
@@ -0,0 +1,70 @@
+package org.wikidata.wdtk.wikibaseapi;
+
+/*-
+ * #%L
+ * Wikidata Toolkit Wikibase API
+ * %%
+ * Copyright (C) 2014 - 2023 Wikidata Toolkit Developers
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import java.util.Objects;
+
+/**
+ * Holds information about a successful edit made via {@link WikibaseDataEditor}.
+ * The state of the entity after edit is not provided here because it is not possible
+ * for WDTK to determine it reliably from the response of the server. Indeed, it is
+ * possible that the data in the entity after the edit differs from the data in the
+ * entity before the edit plus the changes of the edit itself, because it can be that
+ * another edit touched independent parts of the entity. This can happen even if the base
+ * revision id is provided.
+ */
+public class EditingResult {
+    
+    private final long revisionId;
+
+    public EditingResult(long revisionId) {
+        super();
+        this.revisionId = revisionId;
+    }
+
+    /**
+     * The identifier of the revision of the last edit made by the editing action.
+     * If no edit was made, 0L is returned.
+     */
+    public long getLastRevisionId() {
+        return revisionId;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(revisionId);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
+        if (obj == null) return false;
+        if (getClass() != obj.getClass()) return false;
+        EditingResult other = (EditingResult) obj;
+        return revisionId == other.revisionId;
+    }
+
+    @Override
+    public String toString() {
+        return "EditingResult [revisionId=" + revisionId + "]";
+    }
+    
```

Comment:
Just a suggestion: It might be a good idea to use [`EqualsBuilder`](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/EqualsBuilder.html) and [`ToStringBuilder`](https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/ToStringBuilder.html) from Apache Commons (and perhaps their `reflectionEquals` and `reflectionToString` methods) across WDTK code if Java records are too far in the future. It minimizes bugs at the cost of a little performance overhead.

Reply:
I did not know about those. When writing this class I could simply use `@Data` from Lombok:
https://projectlombok.org/features/Data
Overall I think there could be quite some benefits to using Lombok in WDTK (but I have not used it in any real project yet). What do you think?

Reply:
I used Lombok in my projects for some time and then dropped it due to severe compatibility problems that were blocking upgrades. Lombok has to rely on internal compiler APIs and it therefore requires specialized integration with every compiler (and thus every IDE) and sometimes breaks when compiler is upgraded. Plus anyone working on the project has to manually add Lombok to their IDE configuration (see the Install menu on Lombok website). It's less compatible than just using Java 17 and records.

Reply:
Oh great, then that's saving me a lot of effort by not looking into it.